File: examples/tutorial-scala/src/main/scala/tutorial_scala/controller/twitter/IndexController.scala
package tutorial_scala.controller.twitter

import org.slim3.controller.Navigation
import org.slim3scala.controller.Controller

import tutorial_scala.service.TwitterService

class IndexController extends Controller {
  private val service = new TwitterService

  lazy val tweetList = service.getTweetList

  protected def run(): Navigation = {
    forward("index.html")
  }
}

File: examples/tutorial-java/src/main/java/tutorial_java/controller/twitter/IndexController.java
package tutorial_java.controller.twitter;

import java.util.List;

import org.slim3.controller.Controller;
import org.slim3.controller.Navigation;

import tutorial_java.model.Tweet;
import tutorial_java.service.TwitterService;

public class IndexController extends Controller {

    private TwitterService service = new TwitterService();

    @Override
    public Navigation run() throws Exception {
        List<Tweet> tweetList = service.getTweetList();
        requestScope("tweetList", tweetList);
        return forward("index.jsp");
    }
}
